app/path.go

```diff
@@ -228,18 +231,49 @@ func collectSchemaNodes(e *yang.Entry, leafOnly bool) []*yang.Entry {
 		collected = append(collected,
 			collectSchemaNodes(child, leafOnly)...)
 	}
+
 	if e.Parent != nil {
 		switch {
 		case e.Dir == nil && e.ListAttr != nil: // leaf-list
 			fallthrough
 		case e.Dir == nil: // leaf
-			collected = append(collected, e)
+			f := &yang.Entry{
+				Parent:      e.Parent,
+				Node:        e.Node,
+				Name:        e.Name,
+				Description: e.Description,
+				Default:     e.Default,
+				Units:       e.Units,
+				Kind:        e.Kind,
+				Config:      e.Config,
+				Prefix:      e.Prefix,
+				Mandatory:   e.Mandatory,
+				Dir:         e.Dir,
+				Key:         e.Key,
+				Type:        e.Type,
+				Exts:        e.Exts,
+				ListAttr:    e.ListAttr,
+				Extra:       make(map[string][]interface{}),
+			}
+			for k, v := range e.Extra {
+				f.Extra[k] = v
+			}
+			collected = append(collected, f)
 		case e.ListAttr != nil: // list
 			fallthrough
 		default: // container
 			if !leafOnly {
 				collected = append(collected, e)
 			}
+			if e.Extra["if-feature"] != nil {
+				for _, myleaf := range collected {
+					if myleaf.Extra["if-feature"] != nil && myleaf.Extra["if-feature"][0] != nil {
```

Comment:
Shouldn't this check if `len(e.Extra["if-feature"]) != 0` ?
Also, if `myleaf.Extra["if-feature"] == nil`, initialize it.

Reply:
Noted for `len`
If `myleaf.Extra["if-feature"] == nil`, it is defined in the else block, line 273; is it enough ?